te_dp/src/te_tcp_dp.c

```diff
@@ -734,22 +719,33 @@ int curl_socket_cb(CURL *easy, curl_socket_t s, int action, void *user_p, void *
     * If it doesn't work, still leave it. A bug takes about a week to solve
     * :""((
     */
-    te_socket_node_t* socket_ptr_cb = (te_socket_node_t*)socket_p;
```

Comment:
The function is the crux of the changeset. Curl in itself maintains a mapping b/w socket and user-defined socket pointed. 

When a new socket is created, user can associate his custom pointer, to the socket via `curl_multi_assign`. Post this in every callback associated with this socket, curl will give the pointer to the user. This behavior obviates the need for maintaining a hash table. 

Essentially, the previous code, always inserted the node to hash table, but never did a lookup due to the detailed behavior. With this changeset, there is no need to insert as well.

Reply:
@aravindhank11 : Was this impacting the initial burst of sync packets not being sent out so fast ? Can you please explain on the same.

Reply:
No this was an improvement, I attempted to make. However, seems like my understanding was flawed. When multiple requests are to be streamed over the same connection, curl doesn't maintain the sockfd --> pointer mapping, necessitating a hash map. 

Hash map is not needed for a 1 request over 1 connection scenarios whose scenario is anyways fringe. 

To answer your qn, the drop in perf came as a direct result of in curl_socket_timer_cb function:
```
if(timeout_ms == 0)
    timeout_ms = 1
```